proto/pulumi/provider.proto

```diff
@@ -133,11 +133,20 @@ service ResourceProvider {
     // difference (if any) between them. `Diff` should only be called with values that have at some point been validated
     // by a [](pulumirpc.ResourceProvider.Check) call.
     rpc Diff(DiffRequest) returns (DiffResponse) {}
-    // Create allocates a new instance of the provided resource and returns its unique ID afterwards.  (The input ID
-    // must be blank.)  If this call fails, the resource must not have been created (i.e., it is "transactional").
+
+    // `Create` provisions a new instance of the specified [(custom) resource](custom-resources). It returns a
+    // provider-assigned ID for the resource as well as the output properties that arose from the creation properties.
+    // Output properties are typically the union of the resource's input properties and any additional values that were
+    // computed or made available during creation.
+    //
+    // If creation fails, `Create` may return an [](pulumirpc.ErrorResourceInitFailed) error detail explaining why.
+    // Moreover, if `Create` does return an error, it must be the case that the resource was *not* created (that is,
+    // `Create` can be thought of as transactional or atomic).
     rpc Create(CreateRequest) returns (CreateResponse) {}
-    // Read the current live state associated with a resource.  Enough state must be include in the inputs to uniquely
-    // identify the resource; this is typically just the resource ID, but may also include some properties.
+
+    // `Read` reads the current live state associated with a resource identified by the supplied state. The given state
```

Comment:
This one will need a lot of work and is one of the most confusing bits of Pulumi for implementing providers.

More on it here:

https://github.com/pulumi/pulumi/issues/17650

We might have to go ahead and link all the relevant scenarios that invoke Read (two forms of import, refresh, and resource getters).

Reply:
Tracked in #17691